Language/Java/Lexer.x

```diff
@@ -274,6 +274,11 @@ lexicalError p s = L (mkPos p) (Left ("lexical error: " ++ s))
 data L a = L Pos a
   deriving (Show, Eq)
 
+-- we need to know the length of a token to find its end location in the parser,
+-- so we can create sourcespans that end at the end of a token instead of the start of the following token
+data L1 a = L1 Pos Int a
+  deriving (Show, Eq)
```

Comment:
Mit dem Kommentar bin ich nicht ganz zufrieden. Erst einmal würde ich den historischen Aspekt "instead of the start of the following token" weglassen. Das versteht man nur, wenn man die alte Implementierung kennt. Dann würde ich ganz zu Anfang einmal sagen, dass `Pos` die Position des Tokens ist und `Int` die Länge. Außerdem bitte einen [Haddock-Kommentar](https://haskell-haddock.readthedocs.io/en/latest/markup.html) nutzen, am besten mit `{-|` und `-}`.